src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineDeployment.java

```diff
@@ -79,17 +80,19 @@ public void deploy(DeployConfiguration configuration) throws AppEngineException
       arguments.add(configuration.getProject());
     }
 
-    if (configuration.isPromote()) {
+    if (configuration.getPromote() != null) {
       arguments.add("--promote");
+      arguments.add(configuration.getPromote().toString());
     }
 
     if (!Strings.isNullOrEmpty(configuration.getServer())) {
       arguments.add("--server");
       arguments.add(configuration.getServer());
     }
 
-    if (configuration.isStopPreviousVersion()) {
+    if (configuration.getStopPreviousVersion() != null) {
       arguments.add("--stop-previous-version");
+      arguments.add(configuration.getStopPreviousVersion().toString());
```

Comment:
Hmm. Are you sure about that? To me it seems like they should be separate Booleans (since they are distinct flags). Assuming we intend the Booleans to have a 1-to-1 mapping to gcloud flags.